openedx/core/djangoapps/oauth_dispatch/views.py

```diff
@@ -113,17 +127,27 @@ def dispatch(self, request, *args, **kwargs):
 
         return response
 
-    def _decompose_access_token_response(self, request, response):
+    def _decompose_access_token_response(self, adapter, response):
         """ Decomposes the access token in the request to an expiration date, scopes, and User. """
         content = json.loads(response.content)
         access_token = content['access_token']
         scope = content['scope']
-        access_token_obj = self.get_adapter(request).get_access_token(access_token)
-        user = access_token_obj.user
         scopes = scope.split(' ')
+        user = adapter.get_access_token(access_token).user
         expires_in = content['expires_in']
         return expires_in, scopes, user
 
+    def _get_client_specific_claims(self, client_id, adapter):
+        """ Get claims that are specific to the client. """
+        if adapter.is_client_restricted(client_id):
+            issuer_setting = 'RESTRICTED_APPLICATION_JWT_ISSUER'
```

Comment:
So are we going to start signing with the new issuer keys as soon as this PR merges?  Or should this also be behind the feature toggle?
If not, we need to update the config to make sure there's a valid issuer key before releasing this.

Reply:
Good catch. I will put this behind the switch.